service/r2/routes.go

```diff
@@ -64,6 +64,27 @@ func createNamespace(s *service, r *http.Request) (data interface{}, err error)
 	return newNamespaceJSON(view), nil
 }
 
+func validateNamespace(s *service, r *http.Request) (data interface{}, err error) {
+	vars := mux.Vars(r)
+	var rsj ruleSetJSON
+	if err := parseRequest(&rsj, r.Body); err != nil {
+		return nil, err
+	}
+	if vars[namespaceIDVar] != rsj.Namespace {
+		return nil, fmt.Errorf(
+			"NamespaceID param: %s and RuleSet namespaceID: %s do not match",
```

Comment:
nit: error messages should start with lowercase letters. Also might be easier on the eyes to say
`namespaceID param %s and ruleset namespaceID %s do not match`?